packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_model.dart

```diff
@@ -98,25 +195,272 @@ class LoggingTableModel extends ChangeNotifier with DisposerMixin {
+
+    // If the message string was truncated by the VM, or the error object or
+    // stackTrace objects were non-null, we need to ask the VM for more
+    // information in order to render the log entry. We do this asynchronously
+    // on-demand using the `detailsComputer` Future.
+    if (messageRef.valueAsStringIsTruncated == true ||
+        _isNotNull(error) ||
+        _isNotNull(stackTrace)) {
+      detailsComputer = () async {
+        // Get the full string value of the message.
+        String result =
+            await _retrieveFullStringValue(service, e.isolate!, messageRef);
+
+        // Get information about the error object. Some users of the
+        // dart:developer log call may pass a data payload in the `error`
+        // field, encoded as a json encoded string, so handle that case.
+        if (_isNotNull(error)) {
+          if (error!.valueAsString != null) {
+            final errorString =
+                await _retrieveFullStringValue(service, e.isolate!, error);
+            result += '\n\n$errorString';
+          } else {
+            // Call `toString()` on the error object and display that.
+            final toStringResult = await service!.invoke(
+              e.isolate!.id!,
+              error.id!,
+              'toString',
+              <String>[],
+              disableBreakpoints: true,
+            );
+
+            if (toStringResult is ErrorRef) {
+              final errorString = _valueAsString(error);
+              result += '\n\n$errorString';
+            } else if (toStringResult is InstanceRef) {
+              final str = await _retrieveFullStringValue(
+                service,
+                e.isolate!,
+                toStringResult,
+              );
+              result += '\n\n$str';
+            }
+          }
+        }
+
+        // Get info about the stackTrace object.
+        if (_isNotNull(stackTrace)) {
+          result += '\n\n${_valueAsString(stackTrace)}';
+        }
+
+        return result;
+      };
+    }
+
+    const severeIssue = 1000;
+    final isError = level != null && level >= severeIssue ? true : false;
+
+    log(
+      LogData(
+        loggerName,
+        details,
+        e.timestamp,
+        isError: isError,
+        summary: summary,
+        detailsComputer: detailsComputer,
+      ),
+    );
+  }
+
   /// Add a log to the list of tracked logs.
   void add(LogDataV2 log) {
-    // TODO(danchevalier): ensure that search and filter lists are updated here.
+    final newEntry = _LogEntry(log);
+    _logs.add(newEntry);
+    getLogHeight(_logs.length - 1);
+    _trimOneOutOfRetentionLog();
 
-    _logs.add(log);
-    _filteredLogs.add(log);
+    if (!_filterCallback(newEntry)) {
+      // Only add the log to filtered logs if it matches the filter.
+      return;
+    }
+    _filteredLogs.add(_FilteredLogEntry(newEntry));
 
-    _trimOneOutOfRetentionLog();
+    // TODO(danchevalier): Calculate the new offset here
```

Comment:
Are you planning on doing this in a follow up or this PR?

Reply:
yessir offsets will only really be useful with search behaviour so I'm going to nail those down in that PR.